### PR TITLE
Add goreleaser config

### PR DESCRIPTION
**Pull request by sascha-andres**
I added a goreleaser configuration to create packages (tar.gz/deb)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/rdegges/go-ipify"
 )
 
-const (
+var (
```
Comment (robphoenix) [inline main.go:17]: Why the change?

**Comment by robphoenix** [comment]
Thanks @sascha-andres! This looks great.

I'll be able to give this a proper look next week.

File: main.go (hunk 1 above)
Comment (sascha-andres) [reply main.go:17]: Because of `-s -w -X main.Version={{.Version}}` in .goreleaser.yml. Essentially this sets `Version` to either the tag value for a release or the commit sha. So (1) you do not have to set it on your own when releasing, (2) `trawl -v` clearly communicates whether the binary is a snapshot, a released version or built using go install.

To accomplish (2) I will push a commit changing the default to `manual`

Comment (robphoenix) [reply main.go:17]: ah, this is so cool. 💯

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+build: 
+  ldflags: -s -w -X main.Version={{.Version}}
+  binary: trawl
+  env: 
+    - CGO_ENABLED=0 
+  goos:
```
Comment (robphoenix) [inline .goreleaser.yml:6]: Is there a reason Windows isn't included here?

**Comment by robphoenix** [comment]
Sorry for the delay @sascha-andres, I've been pretty sick.

So I understand properly, will this `goreleaser` config also create the binaries I'm currently building with [gox](https://github.com/mitchellh/gox)? (including the windows `.exe` files)

File: .goreleaser.yml (hunk 2 above)
Comment (sascha-andres) [reply .goreleaser.yml:6]: Other that I started based on a template for projects that do not have to run on Windows, no. Pushing an update.

**Comment by sascha-andres** [comment]
No problem @robphoenix 

Yes, it also creates the binaries along creating packages ( I activate debian packages only, as I can not test rpm packages ) and uploading to GITHUB when your token is set.

Maybe the changelog should be customized and/or the release set to be a draft for manual review depending on your needs or likes.

File: .goreleaser.yml (hunk 2 above)
Comment (robphoenix) [reply .goreleaser.yml:6]: ok, thanks, I have to use Windows 7 for work so I like to make sure I can support it as much as possible.

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    - linux
+    - darwin
+    - windows
+  goarch: 
+    - amd64
+archive:
+  format: tar.gz
+snapshot: 
+  name_template: "{{.Commit}}"
+fpm:
```
Comment (robphoenix) [inline .goreleaser.yml]: looks as though this needs to be updated - https://goreleaser.com/deprecations/#fpm

Comment (sascha-andres) [reply .goreleaser.yml]: @robphoenix done

**Comment by robphoenix** [comment]
I've had a hard time getting goreleaser installed and working porperly, and it doesn't look like I can do a trial run without pushing up a new tag. So I'm going to merge this in, as I can see it'll be helpful, and I'll push a new release later this week when I've got more bandwidth available to focus on this.

Thanks @sascha-andres for your effort into this, I really appreciate it. 👍
